fem/adios2datacollection.hpp

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2010, Lawrence Livermore National Security, LLC. Produced at
+// the Lawrence Livermore National Laboratory. LLNL-CODE-443211. All Rights
+// reserved. See file COPYRIGHT for details.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability see http://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the GNU Lesser General Public License (as published by the Free
+// Software Foundation) version 2.1 dated February 1999.
+//
+// Created on: Jan 7, 2020
+// Author: William F Godoy godoywf@ornl.gov
+// adios2: Adaptable Input/Output System https://github.com/ornladios/ADIOS2
+
+
+#ifndef MFEM_ADIOS2DATACOLLECTION
+#define MFEM_ADIOS2DATACOLLECTION
+
+#include "../config/config.hpp"
+#include "../general/adios2stream.hpp"
+#include "datacollection.hpp"
+
+#include <string>
+#include <memory> // std::unique_ptr
+
+namespace mfem
+{
+
+class ADIOS2DataCollection : public DataCollection
```

Comment:
I think this could be added to `datacollection.hpp` instead of its own file. `VisItDataCollection` is done the same way.

Reply:
Or is it because of the Apache license?

Reply:
Good point, I separated mostly to make developers life easier avoiding conflicts and because of the external `adios2` requirement which other datacollections (e.g. visit, paraview depend on `std::`) don't rely on (there are separate conduit and sidre data collection files, I thought that way the criteria). I'm not sure of the implications of the Apache License, as far as I know is very permissive.